packages/zcli-apps/tests/functional/server.test.ts

```diff
@@ -84,4 +85,34 @@ describe('apps server', function () {
         expect(installations[1].updated_at).to.includes(yyyymmdd)
       })
   })
+
+  describe('with manifest.json changes', () => {
+    const wait = (ms = 10) => new Promise(resolve => setTimeout(resolve, ms))
+    let server: http.Server, appHost: string, appJSON: AppJSONPayload
+    before(async () => {
+      server = await ServerCommand.run([singleProductApp])
+      appHost = 'http://localhost:4567'
+    })
+
+    after(() => server.close())
+
+    test
+      .it('should reflect changes in mainfest.json', async () => {
+        // Read mainfest.json
+        const mainfestPath = path.join(singleProductApp, 'manifest.json')
+        const manifest: Manifest = JSON.parse(fs.readFileSync(mainfestPath, 'utf8'))
+        const appName = manifest.name
+        manifest.name = `${appName} modified`
+        // Modifed mainfest.json
+        fs.writeFileSync(mainfestPath, JSON.stringify(manifest))
+        await wait(200)
```

Comment:
Would it be possible to avoid an explicit wait here? Maybe consider an await as prefix to `fs.writeFileSync` on the line above instead?

Reply:
`fs.writeFileSync` returns nothing (void), we can't really `await` as `await` need a promise. I reduced the wait time to `10 ms`.

Reply:
I think the approach here would be to use the promises module in fs, [see this stackoverflow answer](https://stackoverflow.com/a/52094177) for instance.  I remember now, that is why we have those weird imports of the promise submodule as fs in other parts of zcli - it is precisely because of this issue.

So you'd do something like this

```
import { promises as fs } from 'fs';

(async () => {
    [...preceding code...]
    await fs.writeFile(mainfestPath, JSON.stringify(manifest), 'utf8');
    [...following code...]
})();
```

Reply:
Yes, I did `import { promises as fs } from 'fs';` with chokidar but `await wait()` is still required. Because we need to regenerate `app.json` and reset the middlewares  after `manifest.json` changed. Without waiting we get the old `app.json` as resetting the middlewares is done after we fetched `app.json` (based on logs I added).

Reply:
I'm not entirely convinced, but I won't withhold a +1 on this PR if you believe that this is a hard blocker to removing the use of `wait()` here.  `wait` is definitely a smell though which it would be good to expurgate from the code in a follow up task.

Reply:
I've created a followup task and childed it to the current one in the sprint.